service_config.go

```diff
@@ -278,8 +278,7 @@ func convertRetryPolicy(jrp *jsonRetryPolicy, maxAttempts int) (p *internalservi
 		jrp.MaxBackoff <= 0 ||
 		jrp.BackoffMultiplier <= 0 ||
 		len(jrp.RetryableStatusCodes) == 0 {
-		logger.Warningf("grpc: ignoring retry policy %v due to illegal configuration", jrp)
-		return nil, nil
+		return nil, fmt.Errorf("invalid retry policy (%v): ", jrp)
```

Comment:
I know this is existing code, but this seems to violate the style guide and is clearly hard to read given that the return statement is indented at the same width corresponding to the above conditional. See: go/go-style/decisions#conditional-formatting

Could we just move all the conditions into a single line?

Reply:
Also, should we use `pretty.JSON` or a different formatting directive like `%+v` or `%#v`?